README.md

```diff
@@ -34,4 +35,91 @@ qmake
+
+* **holochain-rust** contains the Holochain repository as a sub-module. This project compiles to a static library.
+* **bindings** contains Qt/C++ wrappers around the Holochain Rust code that is made available in the form of plain C functions and structs. This project makes Holochain available as Qt-based QObjects that can easily be plugged into Qt script engines. This project compiles to a static library.
+* **holosqape** contains the QML based GUI project wich compiles to an executable that can be build for all platforms supported by Qt. It manages installed Holochain apps and currently sports a generic UI that creates widgets and buttons for each zome function an app implements automatically. It also includes a JavaScript based console in which apps can be instantiated.
+* **holoconsole** this builds to another executable that basically hosts the same functionality as holosqape but without GUI. Instead it has some compatibility shims to enable running JavaScript libraries such as [Tape](https://github.com/substack/tape). It's main purpose is hosting an environment for app tests, including scenario integration tests.
+* **app-test-scaffold** this is an example showing how to build an app test script to run with holoconsole.
+
+
+
+## App tests
+
+### Binding objects
+In order to test a Holochain app we need a JS script that we run in holoconsole. The script engine adds a global object ```Container``` to the context before loading and running the user script. It currently has the following methods:
+
+* installedApps()
+* instantiate(dnaHash)
+* installApp(filename)
+
+and more as can be seen in [container.h](bindings/container.h).
+
+instantiate() returns an ```App``` object as defined in [app.h](bindings/app.h) with a method
+
+* call(zome_name, capability_name, function_name, parameters)
+
+(amongst others) which calls the specified zome function, blocks, and returns the result.
+
+This enables us to write code like this:
+
+```javascript
+let dnaHash = Container.installedApps()[0]
+let app = Container.instantiate(dnaHash)
+// app is actually a C++ object that wraps the Rust Holochain instance
+app.name() // -> App name defined in DNA
+app.start() // starts the instance's event loop and later will start the network node and DHT (not implement in Rust yet)
+let result = app.call("zome name", "capability name", "function name", "parameters") // runs the zome function which invovles interpreting the app's WASM code
+```
+
+
+### Using Tape testing framework
+To get proper assertions and formatted output we want to use existing JS scripting frameworks. We chose Tape for now because of its mininmal footprint.
+
+In order to build a JS bundle including Tape, we use Webpack. ```app-test-scaffold``` contains a working example which can be run like this:
+
+```
+cd app-test-scaffold
+npm install
+webpack-cli
+```
+You should now find the output in ```app-test-scaffold/dist/bundle.js```.
+Compile holoconsole and run it like this:
```

Comment:
This doesn't work on linux yet.  There are Caps (`HoloSqape` vs `holoscape`)  problems that I couldn't sort out with my limited knowledge of QtCreator.

Reply:
Perhaps this is related to the `HoloSqape.pro.user` file name?  Weird.  I did a Clean, and then it changed to `holosqape.pro.user` so this may have been a vestig.  But, in the mean time, doing Build All results in error messages below, so still can't build.  I did do submodule update and `cargo build` inside the holochain-rust submodule.

```
13:36:29: Running steps for project bindings...
13:36:29: Configuration unchanged, skipping qmake step.
13:36:29: Starting: "/usr/bin/make" 
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o container.o ../bindings/container.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o app.o ../bindings/app.cpp
g++ -pipe -g -fPIC -std=gnu++11 -Wall -W -dM -E -o moc_predefs.h /home/eric/Qt/5.11.1/gcc_64/mkspecs/features/data/dummy.cpp
/home/eric/Qt/5.11.1/gcc_64/bin/moc -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -I/home/eric/code/metacurrency/holochain/holosqape/bindings -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -I/usr/include/c++/5 -I/usr/include/x86_64-linux-gnu/c++/5 -I/usr/include/c++/5/backward -I/usr/lib/gcc/x86_64-linux-gnu/5/include -I/usr/local/include -I/usr/lib/gcc/x86_64-linux-gnu/5/include-fixed -I/usr/include/x86_64-linux-gnu -I/usr/include ../bindings/container.h -o moc_container.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o moc_container.o moc_container.cpp
/home/eric/Qt/5.11.1/gcc_64/bin/moc -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -I/home/eric/code/metacurrency/holochain/holosqape/bindings -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -I/usr/include/c++/5 -I/usr/include/x86_64-linux-gnu/c++/5 -I/usr/include/c++/5/backward -I/usr/lib/gcc/x86_64-linux-gnu/5/include -I/usr/local/include -I/usr/lib/gcc/x86_64-linux-gnu/5/include-fixed -I/usr/include/x86_64-linux-gnu -I/usr/include ../bindings/app.h -o moc_app.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o moc_app.o moc_app.cpp
rm -f libbindings.a
ar cqs libbindings.a container.o app.o moc_container.o moc_app.o
13:36:32: The process "/usr/bin/make" exited normally.
13:36:32: Running steps for project all...
13:36:32: Configuration unchanged, skipping qmake step.
13:36:32: Starting: "/usr/bin/make" 
cd bindings/ && ( test -e Makefile || /home/eric/Qt/5.11.1/gcc_64/bin/qmake -o Makefile /home/eric/code/metacurrency/holochain/holosqape/bindings/bindings.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile 
make[1]: Entering directory '/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings'
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../holosqape/bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o container.o ../../holosqape/bindings/container.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../holosqape/bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o app.o ../../holosqape/bindings/app.cpp
g++ -pipe -g -fPIC -std=gnu++11 -Wall -W -dM -E -o moc_predefs.h /home/eric/Qt/5.11.1/gcc_64/mkspecs/features/data/dummy.cpp
/home/eric/Qt/5.11.1/gcc_64/bin/moc -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -I/home/eric/code/metacurrency/holochain/holosqape/bindings -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -I/usr/include/c++/5 -I/usr/include/x86_64-linux-gnu/c++/5 -I/usr/include/c++/5/backward -I/usr/lib/gcc/x86_64-linux-gnu/5/include -I/usr/local/include -I/usr/lib/gcc/x86_64-linux-gnu/5/include-fixed -I/usr/include/x86_64-linux-gnu -I/usr/include ../../holosqape/bindings/container.h -o moc_container.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../holosqape/bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o moc_container.o moc_container.cpp
/home/eric/Qt/5.11.1/gcc_64/bin/moc -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB --include ./moc_predefs.h -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -I/home/eric/code/metacurrency/holochain/holosqape/bindings -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -I/usr/include/c++/5 -I/usr/include/x86_64-linux-gnu/c++/5 -I/usr/include/c++/5/backward -I/usr/lib/gcc/x86_64-linux-gnu/5/include -I/usr/local/include -I/usr/lib/gcc/x86_64-linux-gnu/5/include-fixed -I/usr/include/x86_64-linux-gnu -I/usr/include ../../holosqape/bindings/app.h -o moc_app.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../holosqape/bindings -I. -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o moc_app.o moc_app.cpp
rm -f libbindings.a
ar cqs libbindings.a container.o app.o moc_container.o moc_app.o
make[1]: Leaving directory '/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings'
cd holosqape/ && ( test -e Makefile || /home/eric/Qt/5.11.1/gcc_64/bin/qmake -o Makefile /home/eric/code/metacurrency/holochain/holosqape/holosqape/holosqape.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile 
make[1]: Entering directory '/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/holosqape'
g++ -c -pipe -g -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../holosqape/holosqape -I. -I../../holosqape/bindings -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o main.o ../../holosqape/holosqape/main.cpp
/home/eric/Qt/5.11.1/gcc_64/bin/rcc -name qml ../../holosqape/holosqape/qml.qrc -o qrc_qml.cpp
g++ -c -pipe -g -std=gnu++11 -Wall -W -D_REENTRANT -fPIC -DQT_DEPRECATED_WARNINGS -DQT_QML_DEBUG -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../../holosqape/holosqape -I. -I../../holosqape/bindings -I/home/eric/Qt/5.11.1/gcc_64/include -I/home/eric/Qt/5.11.1/gcc_64/include/QtQuick -I/home/eric/Qt/5.11.1/gcc_64/include/QtGui -I/home/eric/Qt/5.11.1/gcc_64/include/QtQml -I/home/eric/Qt/5.11.1/gcc_64/include/QtNetwork -I/home/eric/Qt/5.11.1/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I/home/eric/Qt/5.11.1/gcc_64/mkspecs/linux-g++ -o qrc_qml.o qrc_qml.cpp
g++ -Wl,-rpath,/home/eric/Qt/5.11.1/gcc_64/lib -o holosqape main.o qrc_qml.o   -L../../holosqape/holochain-rust/target/debug/ -lholochain_dna_c_binding -lholochain_core_api_c_binding -ldl -L../bindings -lbindings -L/home/eric/Qt/5.11.1/gcc_64/lib -lQt5Quick -lQt5Gui -lQt5Qml -lQt5Network -lQt5Core -lGL -lpthread 
../bindings/libbindings.a(container.o): In function `Container::installApp(QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:23: undefined reference to `holochain_dna_create_from_json'
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:39: undefined reference to `holochain_dna_free'
../bindings/libbindings.a(container.o): In function `getDna(QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:55: undefined reference to `holochain_dna_create_from_json'
../bindings/libbindings.a(container.o): In function `Container::appName(QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:61: undefined reference to `holochain_dna_get_name'
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:63: undefined reference to `holochain_dna_string_free'
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:64: undefined reference to `holochain_dna_free'
../bindings/libbindings.a(container.o): In function `Container::instantiate(QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/container.cpp:109: undefined reference to `holochain_dna_free'
../bindings/libbindings.a(app.o): In function `App::App(QString, QObject*)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:15: undefined reference to `holochain_new'
../bindings/libbindings.a(app.o): In function `App::start()':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:26: undefined reference to `holochain_start'
../bindings/libbindings.a(app.o): In function `App::stop()':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:32: undefined reference to `holochain_stop'
../bindings/libbindings.a(app.o): In function `App::call(QString, QString, QString, QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:38: undefined reference to `holochain_call'
../bindings/libbindings.a(app.o): In function `App::zome_names()':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:48: undefined reference to `holochain_dna_get_zome_names'
../bindings/libbindings.a(app.o): In function `App::capabilities_for_zome(QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:59: undefined reference to `holochain_dna_get_capabilities_names'
../bindings/libbindings.a(app.o): In function `App::functions_for_zome_capability(QString, QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:70: undefined reference to `holochain_dna_get_function_names'
../bindings/libbindings.a(app.o): In function `App::parameter_names(QString, QString, QString)':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:82: undefined reference to `holochain_dna_get_function_parameters'
../bindings/libbindings.a(app.o): In function `App::name() const':
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:95: undefined reference to `holochain_dna_get_name'
/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/bindings/../../holosqape/bindings/app.cpp:97: undefined reference to `holochain_dna_string_free'
collect2: error: ld returned 1 exit status
make[1]: *** [holosqape] Error 1
make: *** [sub-holosqape-make_first] Error 2
Makefile:261: recipe for target 'holosqape' failed
make[1]: Leaving directory '/home/eric/code/metacurrency/holochain/build-all-Desktop_Qt_5_11_1_GCC_64bit-Debug/holosqape'
Makefile:71: recipe for target 'sub-holosqape-make_first' failed
13:36:37: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project all (kit: Desktop Qt 5.11.1 GCC 64bit)
When executing step "Make"
13:36:37: Elapsed time: 00:08.
```

Reply:
Did you build Holochain rust?
I can't find the actual error that ld returns, but it mentions all the functions that include calls into the Rust <> C bindings. Depending on your local configuration, the build might have gone into a different directory.

Also, if you try to build debug, you also need to have the debug Rust build, and release if you want to build release.